crates/goose/src/providers/base.rs

```diff
@@ -317,6 +317,29 @@ pub trait Provider: Send + Sync {
     where
         Self: Sized;
 
+    /// Internal method that performs completion with a specific model
+    /// This is where providers implement their actual completion logic
+    ///
+    /// # Arguments
+    /// * `model` - The model name to use
+    /// * `system` - The system prompt that guides the model's behavior
+    /// * `messages` - The conversation history as a sequence of messages
+    /// * `tools` - Optional list of tools the model can use
+    ///
+    /// # Returns
+    /// A tuple containing the model's response message and provider usage statistics
+    ///
+    /// # Errors
+    /// ProviderError
+    ///   - It's important to raise ContextLengthExceeded correctly since agent handles it
```

Comment:
what are these comments? they repeat the existing comments which are now out of date demonstrating why we shouldn't have them in the first place